helpers/executeCommands.js

```diff
@@ -2,22 +2,15 @@ const executeCommand = require('./executeCommand');
 
 /**
  * A helper function that executes shell commands or JavaScript functions.
- * Supports asynchronous operations.
+ * Supports asynchronous JavaScript functions.
  * Simulates `set -e` behavior in shell, i.e. exits as soon as any commands fail
- * @param  {(string | function(): (number | Promise<number>))[]} commands
+ * @param  {(string | function(): (void | Promise<void>))[]} commands
  * @return Exit code for the last executed command, a non-zero value indicates failure
  */
 module.exports = async function executeCommands(commands) {
   // eslint-disable-next-line no-restricted-syntax
   for (const command of commands) {
     // eslint-disable-next-line no-await-in-loop
-    const code = await executeCommand(command);
-
-    // Return on first non-zero exit value
-    if (code !== 0) {
-      return code;
-    }
+    await executeCommand(command);
```

Comment:
Do the commands have to run in sequence? Should they not be parallelized?

Reply:
`executeCommands` is used for commands that depend on each other, for example: decrypt the secrets, use the secrets to authenticate the CLI tool, use the CLI tool to deploy the app. `executeCommandsInParallel` is used for commands that can be parallelized, like building the client and the server.